link-grammar/parse/histogram.h

```diff
@@ -42,36 +42,37 @@ typedef int64_t count_t;
  *     total == sum_i bin[i] + overrun
  */
 #define NUM_BINS 12
-struct count_t_s
+typedef struct
 {
 	short base;
-	count_t total;
-	count_t bin[NUM_BINS];
-	count_t overrun;
-};
+	w_count_t total;
+	w_count_t bin[NUM_BINS];
+	w_count_t overrun;
+} Count_bin;
 
-typedef struct count_t_s count_t;
+typedef Count_bin w_Count_bin;
 
-count_t hist_zero(void);
-count_t hist_one(void);
+Count_bin hist_zero(void);
+Count_bin hist_one(void);
 
-void hist_accum(count_t* sum, double, const count_t*);
-void hist_accumv(count_t* sum, double, const count_t);
-void hist_prod(count_t* prod, const count_t*, const count_t*);
-void hist_muladd(count_t* prod, const count_t*, double, const count_t*);
-void hist_muladdv(count_t* prod, const count_t*, double, const count_t);
+void hist_accum(Count_bin* sum, double, const Count_bin*);
+void hist_accumv(Count_bin* sum, double, const Count_bin);
+void hist_prod(Count_bin* prod, const Count_bin*, const Count_bin*);
+void hist_muladd(Count_bin* prod, const Count_bin*, double, const Count_bin*);
+void hist_muladdv(Count_bin* prod, const Count_bin*, double, const Count_bin);
 
-static inline count_t hist_total(count_t* tot) { return tot->total; }
-count_t hist_cut_total(count_t* tot, int min_total);
+static inline w_count_t hist_total(Count_bin* tot) { return tot->total; }
+w_count_t hist_cut_total(Count_bin* tot, count_t min_total);
 
-double hist_cost_cutoff(count_t*, int count);
+double hist_cost_cutoff(Count_bin*, count_t count);
 
 #else
 
 typedef int32_t count_t;
```

Comment:
this is duplicated above.

Reply:
Only the first one is needed. I will remove it in the next cleanup.

Reply:
PR #1273.